### ISSUE TITLE
[Bug] AutomationProperties on Button still visible on Android if Text Property is empty

### ISSUE THREAD

**Issue by MathewLC** [xamarin/Xamarin.Forms#7875]
### Description
This error is hapenning in the actual version of xamarin forms.

### Basic Information

- IDE: Visual Studio 2019
- Platform Target Frameworks:
  - Android: Android 10 (API Level 29)

- Nuget Packages: Xamarin.Forms (4.2.0.848062)
- Tested Devices: Emulators with Android 10 and 9. Nokia x6 with Android 9.

### Screenshots
![image](https://user-images.githubusercontent.com/37350787/66404594-2331c380-e9bf-11e9-8854-7c2ee53be20a.png)

### Reproduction Link
[AppTeste.zip](https://github.com/xamarin/Xamarin.Forms/files/3703015/AppTeste.zip)

**Comment by jfversluis** [comment]
Seems to happen indeed, only on Android. What I somehow suspect is that we somehow, somewhere assign this to a visible property whenever that is empty? Need to investigate a bit on that.

As a workaround you could enter a space in your `Text` property for now.

**Comment by magoolation** [comment]
I'd like to work om this as I'm focused on accessibility improvements on Xamarin.Forms

**Comment by jfversluis** [comment]
By all means, please go ahead! I'll assign it to me for now so no one will snatch it from you. If, for some reason, you can't finish it please let me know :)

**Comment by MathewLC** [comment]
If help you guys, I run into aother issue related to this. It seems like the properties are only hide but definily is on the UI. In the sample I'll send, the same button with and without AutomationProperties, has different width value as you can see in the screenshot.

### ScreenShot
![image](https://user-images.githubusercontent.com/37350787/66483991-80d81580-ea7c-11e9-8316-828dcdf3dfdc.png)


### Reproduction Link
[AppTeste.zip](https://github.com/xamarin/Xamarin.Forms/files/3707456/AppTeste.zip)

**Comment by jfversluis** [comment]
I'm not sure what you're trying to say. You mean using the AutomationProperties are impacting the width?

**Comment by hartez** [comment]
> I'm not sure what you're trying to say. You mean using the AutomationProperties are impacting the width?

We had a problem like that last year; maybe this is a similar situation? https://github.com/xamarin/Xamarin.Forms/issues/2580

**Comment by MathewLC** [comment]
Exactly. In the code that's in the sample I've sended, you can see that the two buttons that reproduce this images, are equals with the only difference being the automationproperties in the last one.

See the folling code:

```
        <!--This don't have extra width because don't have AutomationProperties --> 
        <Button Text="Tiny"
                VerticalOptions="CenterAndExpand"
                HorizontalOptions="Center"
                BackgroundColor="Blue"/>
    
        <!-- This have extra width because of AutomationProperties -->
        <Button Text="Tiny" 
                AutomationProperties.Name="Name"
                AutomationProperties.HelpText="Help Large Text "
                VerticalOptions="CenterAndExpand"
                HorizontalOptions="Center"
                BackgroundColor="Blue"/>

**Comment by hartez** [comment]
I wonder if this is showing up again because the fast renderers are the default now? Looks like Rui fixed this in the older LabelRenderer; maybe the same fix needs to be applied to the fast renderer: https://github.com/xamarin/Xamarin.Forms/pull/4433/files#diff-c754c889f501d0bbd657ea1ca0ad3370

**Comment by jfversluis** [comment]
Nice catch @hartez!

@magoolation this should be an easy fix then :) are you able to implement this anytime soon? Else to get this bug out of the way ASAP I can do it for you

**Comment by magoolation** [comment]
My plan is to do it next week as I’m attending to a conference until Sunday

**Comment by jfversluis** [comment]
@magoolation Not trying to put pressure on you, sorry! But did you make any progress with this yet? Please let us know if you need help or want us to take over. Thanks!

**Comment by magoolation** [comment]
Thank you for asking.
I couldn’t find a solution yet
If you can give me a direction to follow i will aprecia-te it
But if you think it will be easier solve it by yourself go ahead

**Comment by jfversluis** [comment]
Did you see this comment from a little bit earlier? You can find it here: https://github.com/xamarin/Xamarin.Forms/issues/7875#issuecomment-540141157

Although I do see they are talking about a `Label` and not a `Button` there... Maybe I need to do some more research myself :)

**Comment by MathewLC** [comment]
I didn't test the label. But I'm sure the button error was fixed in old render.
It seems like this is the right commit in which the fix was done:
https://github.com/xamarin/Xamarin.Forms/pull/4094/commits/ff13c49b53bb258e824d9e51cd2f6378a5d227c1 @jfversluis @magoolation 
> Did you see this comment from a little bit earlier? You can find it here: [#7875 (comment)](https://github.com/xamarin/Xamarin.Forms/issues/7875#issuecomment-540141157)
> 
> Although I do see they are talking about a `Label` and not a `Button` there... So this code might be needed in the `LabelRenderer` in the FastRenderers folder but also on the button?

**Comment by jfversluis** [comment]
Thanks @MathewLC !

Let me know if you can make any progress with this @magoolation. Else I will have a go at this to get this a11y bug out of the way!

**Comment by magoolation** [comment]
Thank you for the help. I'll give it a try tonight and let you know if I
need something else.

Em seg, 28 de out de 2019 às 09:46, Gerald Versluis <
notifications@github.com> escreveu:

**Comment by czuck** [comment]
I can confirm that this is happening with Labels as well.  Here is my test code:
```
        <StackLayout
            VerticalOptions="FillAndExpand"
            Orientation="Vertical"
            BackgroundColor="Yellow"
            Spacing="10" >
            <Label x:Name="FirstLabel"
                Text="First Label"
                HorizontalOptions="Center"
                Font="12" 
                   BackgroundColor="Aqua"/>
            <Label x:Name="SecondLabel"
                Text="Second Label"
                HorizontalOptions="Center"
                Font="12" 
                   BackgroundColor="BlueViolet"
            AutomationProperties.Name="This is a really long Automation Properties Name, It's so long it takes up two lines"/>
            <Label x:Name="ThirdLabel"
                Text="Third Label"
                HorizontalOptions="Center"
                Font="12" 
                   BackgroundColor="ForestGreen"/>
            <Label x:Name="FourthLabel"
                Text="Fourth Label"
                HorizontalOptions="Center"
                Font="12" 
                   BackgroundColor="HotPink"
                   AutomationProperties.Name="Short Name"/>
        </StackLayout>
```
```
        public MainPage()
        {
            InitializeComponent();
            AutomationProperties.SetName(FirstLabel, "This is a really long Automation Properties Name, It's so long it takes up two lines");
            AutomationProperties.SetName(ThirdLabel, "Short Name");
        }
```
![image](https://user-images.githubusercontent.com/1641428/69381425-99397380-0c69-11ea-9e4e-449dfee8df01.png)

**Comment by MathewLC** [comment]
@czuck I think you can wait and see if the pull request that reference this bug fix it in the next version of xamarin forms(4.4.0). If not you can open another Issue and reference this.

**Comment by czuck** [comment]
Labels are working properly with 4.4.0-pre3!  Thanks!

**Comment by jfversluis** [comment]
Thanks for letting us know @czuck ! @MathewLC can you confirm and update this issue appropriately?

**Comment by MathewLC** [comment]
@jfversluis  I tested with the last release version of forms and in fact labels are working but buttons aren't. You guys didn't merge this yet, am I right?

**Comment by MathewLC** [comment]
@jfversluis  I tested with the last release version of forms and in fact labels are working but buttons aren't. You guys didn't merge this yet, am I right?

**Comment by jfversluis** [comment]
Whoops, no the PR is still open and I see it was just waiting for my review 😱 sorry about that! I'll see to it that I do that ASAP. Thanks for the ping @MathewLC !

**Comment by samhouts** [comment]
closed by #8368
